### PR TITLE
Hide old deployments in the topology view

### LINKED ISSUE #5091: Old deployment shown in topology view, but not overview
We hide replication controllers that have no pods in the overview, but show them in the topology view. This might be confusing for users since it's inconsistent. Also, there could be dozens of old deployments, resulting in a busy graph.

/cc @stefwalter @jwforres @benjaminapetersen

### PR DESCRIPTION
**Pull request by spadgett**
Fixes #5091

/cc @jwforres @stefwalter

### CODE REVIEW

File: assets/app/scripts/controllers/overview.js [hunk 1]
```diff
@@ -406,6 +406,32 @@ angular.module('openshiftConsole')
         topologyItems[makeId(service)] = service;
       });
 
+      var isRecentDeployment = $filter('isRecentDeployment');
+      $scope.isVisibleDeployment = function(deployment) {
+        // If this is a replication controller and not a deployment, then it's visible.
+        var dcName = annotationFilter(deployment, 'deploymentConfig');
+        if (!dcName) {
```
Comment (jwforres) [inline assets/app/scripts/controllers/overview.js]: any reason not to do  `if ( $filter('isDeployment', deployment) )`

**Comment by jwforres** [comment]
just the one question, otherwise LGTM

File: assets/app/scripts/controllers/overview.js (hunk 1 above)
Comment (spadgett) [inline assets/app/scripts/controllers/overview.js]: Only reason is to avoid resolving the filter inside a loop.

**Comment by jwforres** [comment]
OK.  [merge]

**Comment by spadgett** [comment]
[test]

**Pull request merged**